core/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -102,19 +94,21 @@
 
   private synchronized void appendCurrentBuffer(byte[] buf, int offset, int length)
       throws IOException {
-    if (!mTachyonFS.requestSpace(length)) {
-      mCanWrite = false;
-
-      String msg =
-          "Local tachyon worker does not have enough " + "space (" + length + ") or no worker for "
-              + mFile.mFileId + " " + mBlockId;
-
-      throw new IOException(msg);
+    if (mAvailableBytes < length) {
+      long bytesRequested = mTachyonFS.requestSpace(mBlockId, length - mAvailableBytes);
```

Comment:
this logic means if there is no enough space, request space from worker. if request success mAvailableBytes increases, if request fails, throw Exception.
return value of mTachyonFS.requestSpace:
0 means request fails
-1 means no local worker which also means fails
the requested size means success. and the requested size is the bigger one of length - mAvailableBytes and UserQuotaBytes.